### PR TITLE
Use a proper coordinator table schema based on whether the group commit feature is enabled or not

### PR DESCRIPTION
**Pull request by komamitsu**
## Description

We added the group commit feature for the coordinator table commit in version `3.13` and `4.x`. It added a new column `tx_child_ids` in `coordinator.state` table. So, our plan was as follows:
- In version 3, users manually need to drop and re-create the coordinator table if they want to use the group commit feature
- In version 4, users can migrate their coordinator table to the new schema that contains `tx_child_ids` using `schema-loader --upgrade` command

But, we noticed `schema-loader --repair-all --coordinator` command with JDBC storage updates the ScalarDB metadata table by adding `tx_child_ids` column while the existing coordinator table in underlying RDB is left as-is. It results in the inconsistent state and will cause an issue that `schema-loader --upgrade` wouldn't work with version 4.

After some discussions, we changed our plan for version 3 as follows:

- If the group commit feature is disabled (default), the old coordinator table schema is used. No issue happens even if `schema-loader --repair-all` is executed. So, most users don't need to consider the issue.
- If the group commit feature is enabled, users need to drop and re-create their coordinator table if it exists in advance. 
**This must be documented in another PR later**

## Related issues and/or PRs

None

## Changes made

- Made `ConsensusCommitAdmin` use the existing coordinator table schema that doesn't have `tx_child_idx` column if the group commit feature is disabled, and use the new coordinator table schema that has `tx_child_idx`  column if the group commit feature is enabled
- Updated and added related unit tests and integration tests

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

None

## Release notes

N/A

### CODE REVIEW

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/ConsensusCommitAdminIntegrationTestWithCassandra.java [hunk 1]
```diff
@@ -17,4 +19,20 @@ protected String getSystemNamespaceName(Properties properties) {
...
+  }
+
+  @Override
+  protected boolean isGroupCommitEnabled(String testName) {
+    return new ConsensusCommitConfig(new DatabaseConfig(getProperties(testName)))
+        .isCoordinatorGroupCommitEnabled();
+  }
+
+  @Override
+  protected void extraCheckOnCoordinatorTable() {}
```
Comment (komamitsu) [inline core/src/integration-test/java/com/scalar/db/storage/cassandra/ConsensusCommitAdminIntegrationTestWithCassandra.java:37]: The default check doesn't work. Maybe it's related to https://github.com/scalar-labs/scalardb/blob/f57fe48151915819ea659646cb9406d2e6c2b18d/core/src/integration-test/java/com/scalar/db/storage/cassandra/CassandraAdminRepairIntegrationTest.java#L26-L29...?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you

**Pull request merged**